cmd/thanos/tools_bucket.go

```diff
@@ -203,6 +204,8 @@ func (tbc *bucketWebConfig) registerBucketWebFlag(cmd extkingpin.FlagClause) *bu
 	cmd.Flag("timeout", "Timeout to download metadata from remote storage").Default("5m").DurationVar(&tbc.timeout)
 
 	cmd.Flag("label", "External block label to use as group title").StringVar(&tbc.label)
+
+	cmd.Flag("disable-admin-operations", "Restrict access to Block Mark deletion and no compaction").Default("false").BoolVar(&tbc.disableAdminOperations)
```

Comment:
Shouldn't this flag be in Compactor (`cmd/thanos/compact.go`) instead? As that is the component that exposes blocks UI

Reply:
I lack on a broader knowledge of codebase, but as far as I can understand, If i add the flag in compact.go then the flag will be available in Compactor UI instead of Bucket UI. I can tell this for sure that adding the flag in tools_bucket.go is only affecting Bucket UI.

Reply:
I lack on a broader knowledge of codebase, but as far as I can understand, If i add the flag in compact.go then the flag will be available in Compactor UI instead of Bucket UI. I can tell this for sure that adding the flag in tools_bucket.go is only affecting Bucket UI.

Reply:
Ack got confused for a bit. So Thanos users can access the Block UI via two ways I think,
- `tools bucket web` (i.e which is covered by the flag you added)
- `compact` (which isn't covered)

So let's add the same flag there too, as you can mark deletion and no compaction there too.